astropy/cosmology/tests/test_cosmology.py

```diff
@@ -1577,6 +1577,21 @@ def test_z_at_value():
     assert allclose(z_at_value(cosmo.distmod, 46 * u.mag),
                     1.9913891680278133, rtol=1e-6)
 
+    # Test array:
+    assert allclose(z_at_value(cosmo.distmod, np.array([46]*4) * u.mag,
+                               zmin=1.5, zmax=2.5, nbins=10000),
+                    1.9913891680278133, rtol=1e-3)
+    assert allclose(z_at_value(cosmo.distmod, np.array([46]*4) * u.mag,
+                               zmin=1.5, zmax=2.5, nbins=10000,
+                               logspace=False, interpolation='linear'),
+                    1.9913891680278133, rtol=1e-3)
+    c = core.WMAP9
+    assert allclose(z_at_value(c.distmod, np.linspace(46, 47, num=5)*u.mag,
```

Comment:
good test. should just have a comment explaining purpose. Maybe
``# Test evaluation of an array element-wise matches an array of evaluations``